src/api/utils/filter_functions.py

```diff
@@ -0,0 +1,366 @@
+import json
+import operator
+from functools import reduce
+from django.db.models import Q
+
+
+def get_events_filter_params(params):
```

Comment:
these queries look so complicated and expensive.  We will have to review them thoroughly.

Reply:
@Opoku-Agyemang - what is the best way to review this?

Reply:
I tested this for Actions testing locally but fetching from Dev.  The additional time for applying the filter was 0.2 ms, which is small.  Prod is a much larger DB, so if it is 1000 times larger it may add 0.2 seconds.  We'll need to test this on Prod.